Models/Scraper/Scraper.php

```diff
@@ -134,6 +135,47 @@ private function getCombinedTags($content, array $basicTags)
 		return array_keys($tags);
 	}
 
+	/**
+	 * Get the website description.
+	 * @param string $url Website URL
+	 * @return string Description website
+	 * Resource:
+	 * - https://www.codespeedy.com/get-meta-tags-of-a-web-page-in-php/
+	 * - https://stackoverflow.com/questions/7454644/how-to-get-open-graph-protocol-of-a-webpage-by-php
+	 * Limitations:
+	 * - Shopify sites -> Shopify prevents scraping
+	 * - Sites using React-helmet
+	 */
+	private function getWebsiteDescription($url)
+	{
+		// To get meta description and twitter description
+		$metas = get_meta_tags($url);
```

Comment:
get_meta_tags download the page and parse **only** meta tags with **name** attributes.
It is maybe better to use the DOMDocument class like that:

```
private function getWebsiteDescription($url)
{
  $html = file_get_contents($url);  // Download the HTML page
  $position = strpos($html, '</head>');

  if ($position == false)  // or 0
  { return ''; }  // We can not do anything without the <head> part

  // Remove the <body> part to only parse the <head> (optimization)
  $html = substr($html, 0, $position) . '</head><body></body></html>';
  $dom = new DOMDocument();
  $dom->loadHTML($html);

  $metas = [];

  foreach ($dom->getElementsByTagName('meta') as $meta)
  {
    if ($meta->hasAttribute('name') && strtolower($meta->getAttribute('name')) == 'description')
    { $metas['description'] = trim($meta->getAttribute('content')); }

    if ($meta->hasAttribute('property') && strtolower($meta->getAttribute('property')) == 'og:description')
    { $metas['og:description'] = trim($meta->getAttribute('content')); }

    if ($meta->hasAttribute('name') && strtolower($meta->getAttribute('name')) == 'twitter:description')
    { $metas['twitter:description'] = trim($meta->getAttribute('content')); }
  }

  // Now we can choose which description we prefer if we found many!

  if (!empty($metas['description']))
  { return $metas['description']; }

  if (!empty($metas['og:description']))
  { return $metas['description']; }

  if (!empty($metas['twitter:description']))
  { return $metas['description']; }

  // If there were no description available maybe we can use the page title...
  $title = $dom->getElementsByTagName('title')->item(0);

  if ($title != null && !empty($title->nodeValue))
  { return trim($title->nodeValue); }

  return '';
}
```